contrib/devtools/symbol-check.py

```diff
@@ -51,9 +51,10 @@
 '_edata', '_end', '__end__', '_init', '__bss_start', '__bss_start__', '_bss_end__', '__bss_end__', '_fini', '_IO_stdin_used', 'stdin', 'stdout', 'stderr',
 'environ', '_environ', '__environ',
 }
-CPPFILT_CMD = os.getenv('CPPFILT', '/usr/bin/c++filt')
-OBJDUMP_CMD = os.getenv('OBJDUMP', '/usr/bin/objdump')
-OTOOL_CMD = os.getenv('OTOOL', '/usr/bin/otool')
+
+CPPFILT_CMD = lambda: determine_wellknown_cmd('CPPFILT', 'c++filt')
```

Comment:
What is the rationale for making these lambdas instead of evaluating them here and now, once?

Reply:
In order to appease the python type linter, I made `determine_wellknown_cmd` simply error out if the program is not found

https://github.com/bitcoin/bitcoin/blob/05f870018c618167c4740e7d1381e98bd30ae5f9/contrib/devtools/utils.py#L14-L22

Therefore, if we were to evaluate these immediately, and the tool does not exist in the environment (e.g. `otool` won't exist in a build container for Linux), then the script will just crash.

I'm more than happy to adopt some other, better solution. This was just _a_ solution I found which works

Reply:
I personally think  using lambdas here is not a nice construction. If it is only 'to appease the type linker', please add an ignore pragma `# type: ignore`. Our purpose for linters is to make the code better not unnecessarily verbose :smile: